src/pages/vue/pwa.md

```diff
@@ -1,10 +1,215 @@
 ---
+title: "Progressive Web Apps in Vue"
 previousText: 'Platform Utils'
 previousUrl: '/docs/vue/platform'
 nextText: 'Testing'
 nextUrl: '/docs/vue/testing'
 ---
 
-# Progressive Web Apps in Vue
+## Making your Vue app a PWA
 
-Coming Soon!
\ No newline at end of file
+The two main requirements of a PWA are a <a href="https://developers.google.com/web/fundamentals/primers/service-workers/" target="_blank">Service Worker</a> and a <a href="https://developers.google.com/web/fundamentals/web-app-manifest/" target="_blank">Web Manifest</a>. While it's possible to add both of these to an app manually, the Vue CLI has some utilities for adding this for you.
+
+For existing projects, you can run the `vue add` command to install the PWA plugin for Vue.
+
+```shell
+vue add pwa
+```
+
+> If you have changes already in place, be sure to commit them in Git.
+
+Once this is completed, Vue's CLI will have created a new `registerServiceWorker.ts` file and imported it into our `main.ts`.
+
+
+```typescript
+import { createApp } from 'vue'
+import App from './App.vue'
+import router from './router'
+// Added by the CLI
+import './registerServiceWorker'
+
+createApp(App)
+  .use(router)
+  .mount('#app')
+```
+
+The `registerServiceWorker.ts` will point to a service worker that the CLI will create at build time. Inside of here we can customize the experience users will have when the service worker detects and update, change in network connectivity, or receives an error.
+
+```typescript
+import { register } from 'register-service-worker'
+
+if (process.env.NODE_ENV === 'production') {
+  register(`${process.env.BASE_URL}service-worker.js`, {
+    ready () {
+      console.log(
+        'App is being served from cache by a service worker.\n' +
+        'For more details, visit https://goo.gl/AFskqB'
+      )
+    },
+    registered () {
+      console.log('Service worker has been registered.')
+    },
+    cached () {
+      console.log('Content has been cached for offline use.')
+    },
+    updatefound () {
+      console.log('New content is downloading.')
+    },
+    updated () {
+      console.log('New content is available; please refresh.')
+    },
+    offline () {
+      console.log('No internet connection found. App is running in offline mode.')
+    },
+    error (error) {
+      console.error('Error during service worker registration:', error)
+    }
+  })
+}
+```
+
+The service worker that is generated is based on [Workbox's webpack plugin](https://developers.google.com/web/tools/workbox/modules/workbox-webpack-plugin), and by default is setup to use the `GenerateSW()`. Meaning that at build time, Workbox will automatically generate a service worker cache for all the files it processes.
+
+If you wanted to configure this and change the default behavior, checkout the [PWA plugin docs](https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-pwa#configuration) on GitHub.
+
+
+### Manifest
```

Comment:
Ionic starters ship with a manifest.json file already. Should we remove it from the starters? Ionic React starters ship with them too: https://github.com/ionic-team/starters/tree/master/react/base/public

Reply:
React ships with them but they have service workers unregistered by default. The starters should follow the angular approach of "Assume no pwa, and guide on how to add them"

Reply:
Makes sense - I can remove the manifest.json from the vue starters